staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -108,6 +116,9 @@ func RegisterStorageCleanup(fn func()) {
 }
 
 func CleanupStorage() {
+	if atomic.AddInt64(&trackStorageCleanupAtomic, -1) != 0 {
+		return
```

Comment:
this is called from a single place... is there a reason we wouldn't change the teardown func returned from StartTestServer() to let you specify whether you wanted to tear down storage or not?

Reply:
If we were to change StartTestServer() to pass the option to tear down the storage, then the application logic would need to keep track of which Server 'owned' the tear down. For instance, if there are 4 servers, then it would make sense to add the boolean to the 4th instance. If the 4th instance needed to be destroyed (for testing purposes) then the 4th instance would mess up the other 3 running servers making the code more brittle than just using an atomic.

Reply:
Straw man, add another tear down function to the TestServer:

```go
// TearDownFunc is to be called to tear down a test server.
type TearDownFunc func()

// TestServer return values supplied by kube-test-ApiServer
type TestServer struct {
        ClientConfig *restclient.Config        // Rest client config
        ServerOpts   *options.ServerRunOptions // ServerOpts
        TearDownFn   TearDownFunc              // TearDown function
        TmpDir       string                    // Temp Dir used, by the apiserver

        // TearDownServerFn is similar to TearDownFn, but doesn't clear the API server's
        // backing storage. This should be used when running multiple test API servers,
        // and lets the test bring down individual servers without clearing the backing
        // data.
        TearDownServerFn TearDownFunc
}
```